src/client/api/object.c

```diff
@@ -86,6 +86,9 @@ daos_obj_punch(daos_handle_t oh, daos_handle_t th, uint64_t flags,
 	tse_task_t	*task;
 	int		rc;
 
+	if (!daos_handle_is_inval(th))
```

Comment:
all those changes in this file need to move to the dc_ layer because if we use the task API, those don't get called.

Reply:
I am not sure I can catch your idea. What means for the "task API"? will daos_obj_punch() NOT be called under "task API"?

Reply:
we can call the API through daos_task_create(DAOS_OPC_OBJ_PUNCH, ..) for example, vs daos_obj_punch_dkeys().
that path does not use the API and creates a task directly with the dc_ calls.
see src/include/daos_task.h